### PR TITLE
Admin sales routes

### LINKED ISSUE #4791: Admin: Sales: All but one tabs show demo data
**Is your feature request related to a problem? Please describe.**
Except for `invoices`, all sales tabs display demo data. Please check if the correct endpoints exist. This is present in the tabs

- By Events
- By Organizer
- By Location
- By Marketer
- Discounted events
- Fees

Hint: frontend integration will be necessary

### PR DESCRIPTION
**Pull request by maxlorenz**
Approaches #4791 and implements the 5 routes that calculate sales total and ticket counts

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:

- Adds admin sales routes

#### Changes proposed in this pull request:

- [x] By Events route
- [x] By Organizer
- [x] By Location
- [x] By Marketer
- [x] Discounted events

### CODE REVIEW

File: app/api/admin_sales/marketer.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    email = fields.String()
+    sales = fields.Method('calc_sales')
+
+    @staticmethod
+    def calc_sales(obj):
+        """
+        Returns sales (dictionary with total sales and ticket count) for
+        placed, completed and pending orders
+        """
+        return summary(obj.orders)
```
Comment (Kreijstal) [inline app/api/admin_sales/marketer.py]: Where does the `orders` attribute come from, if there isn't an order attribute in AdminSalesByMarketerSchema?

Comment (maxlorenz) [reply app/api/admin_sales/marketer.py]: From the join [here](https://github.com/maxlorenz/open-event-server/blob/a748d686e6ece5b9621f8c131d88989ba5d55951/app/api/admin_sales/marketer.py#L48)

**Comment by maxlorenz** [comment]
I will finish the PR here so it can be merged now once reviewed. The queries for the location and the marketer are a bit too complicated for my taste but I haven't found a good way to simplify them so far

File: app/api/admin_sales/events.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+from app.api.bootstrap import api
+from app.models import db
+from app.models.event import Event
+from app.models.order import Order, OrderTicket
+
+from app.api.admin_sales.utils import summary
+
+
+class AdminSalesByEventsSchema(Schema):
```
Comment (Kreijstal) [inline app/api/admin_sales/events.py]: Isn't this already extremely similar to https://github.com/fossasia/open-event-server/blob/6b1f5db3ed3c8cf1781dd09830bcb6ba60135205/app/api/admin_statistics_api/events.py#L15 ?

Comment (bhaveshAn) [reply app/api/admin_sales/events.py]: `AdminSalesByEventsSchema `
`AdminStatisticsEventSchema` 
class names differs. @Kreijstal

Comment (maxlorenz) [reply app/api/admin_sales/events.py]: And they do totally different things

File: app/models/order.py [hunk 3]
```diff
@@ -107,10 +107,7 @@ def invoice_number(self):
         return self.get_invoice_number()
 
     def get_tickets_count(self):
-        count = 0
-        for order_ticket in self.order_tickets:
-            count += order_ticket.quantity
-        return count
+        return sum([t.quantity for t in self.order_tickets])
```
Comment (schedutron) [inline app/models/order.py]: Where will the `get_tickets_count()` method be called?

Comment (schedutron) [inline app/models/order.py]: You will add the `tickets_count` property corresponding to this method, just like with the case of `invoice_number`, right?

Comment (maxlorenz) [reply app/models/order.py]: done :)

**Comment by bhaveshAn** [comment]
@maxlorenz Please solve multiple migration heads error using https://blog.fossasia.org/migrations-makes-us-crazy/

**Comment by bhaveshAn** [comment]
@maxlorenz any update on this

**Comment by Kreijstal** [comment]
Hi max, I'm taking the freedom of editing your PR because I want it merged, and you're probably busy with other things.

**Comment by maxlorenz** [comment]
@Kreijstal thanks, I guess only the multiple migration heads error is tbd

**Comment by maxlorenz** [comment]
The multiple heads check script threw an error that was unrelated (PR does not contain migrations). CI should go through now

**Pull request merged**